### PR TITLE
[stable/redis-ha] Add antiAffinity rules

### PR DESCRIPTION
**Pull request by nickgryg**
**What this PR does / why we need it**:

Adding antiAffinity rules for both `redis-server` and `redis-sentinel` pods

### CODE REVIEW

**Comment by davidkarlsen** [comment]
@nickgryg want to merge with master to fix conflicts?

**Comment by davidkarlsen** [comment]
@nickgryg PTAL?

File: stable/redis-ha/templates/redis-sentinel-deployment.yaml [hunk 1]
```diff
@@ -23,6 +23,27 @@ spec:
...
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - weight: 1
+              podAffinityTerm:
+                topologyKey: kubernetes.io/hostname
+                labelSelector:
+                  matchLabels:
+                    app: {{ template "redis-ha.name" . }}
+                    release: {{ .Release.Name }}
```
Comment (kimxogus) [inline stable/redis-ha/templates/redis-sentinel-deployment.yaml]: I think `component` label should be also. What do you think?

**Comment by nickgryg** [comment]
@davidkarlsen I've fixed merge conflicts. Please take a look to the updated PR.

**Pull request merged**